src/move/econia/sources/market.move

```diff
@@ -6390,6 +6834,86 @@ module econia::market {
                        quote_coins);
         // Assert order book counter.
         assert!(get_order_book_counter(market_id) == 3, 0);
+        let taker_order_id = order_id_no_post(3);
+        // Assert events.
+        assert!(user::get_place_limit_order_events_test(
```

Comment:
TBH I would just have a helper for asserting all events given the market id, account address (optional?), and custodian id being asserted. Then you would just pass your expected vectors in and it would assert all of the event types. This way we know for a fact that we've asserted all events that get emitted and all events that don't get emitted--by using such a helper.

Reply:
If a similar logical flow were taking place during runtime I would agree with the recommendation for abstraction

But since these are tests I lean toward using the least sophisticated methods possible e.g. keeping the code dumb and procedural

This way the abstracted test helpers don't themselves also need to be tested